lisa/transformers/kernel_source_installer.py

```diff
@@ -114,23 +119,59 @@ def install(self) -> str:
         # modify code
         self._modify_code(node=node, code_path=code_path)
 
-        self._build_code(node=node, code_path=code_path)
+        config_path = source.get_config_path()
+        self._build_code(node=node, code_path=code_path, kconfig_path=config_path)
 
         self._install_build(node=node, code_path=code_path)
 
         result = node.execute(
-            "make kernelrelease 2>/dev/null", cwd=code_path, shell=True
+            "make kernelrelease 2>/dev/null",
+            cwd=code_path,
+            shell=True,
         )
 
         kernel_version = result.stdout
-        result.assert_exit_code(0, f"failed on get kernel version: {kernel_version}")
+        result.assert_exit_code(
+            0,
+            f"failed on get kernel version: {kernel_version}",
+        )
 
         # copy current config back to system folder.
         result = node.execute(
-            f"cp .config /boot/config-{kernel_version}", cwd=code_path, sudo=True
+            f"cp .config /boot/config-{kernel_version}",
+            cwd=code_path,
+            sudo=True,
         )
         result.assert_exit_code()
 
+        if config_path and len(config_path) > 0:
```

Comment:
`if config_path` should be enough to check None or Empty string. But how the `config_path` relates to dom0? It's not straightforward on names and the logic.

Reply:
This is to check if any config_path is getting passed from runbook. If it is , then we will refer it otherwise we will follow the default flow of referring the config from /boot.

config_path is the path where our config file is located. Ex. for dom0, it is arch/x86/configs/mshv_defconfig under the linux-dom0 repo.

I will remove the length condition.

Reply:
Removed the length condition